cineast-core/src/main/java/org/vitrivr/cineast/core/iiif/imageapi/ImageRequest.java

```diff
@@ -0,0 +1,147 @@
+    this.baseUrl = baseUrl;
+    this.region = region;
+    this.size = size;
+    this.rotation = rotation;
+    this.quality = quality;
+    this.extension = extension;
+  }
+
+  /**
+   * Percent encodes "/","?","#","[","]","@" and "%" to their corresponding ASCII reserved characters
+   */
+  public static String percentEncode(String toEncode) {
+    HashMap<Character, String> mapping = new HashMap<>();
+    mapping.put('/', "%2F");
+    mapping.put('?', "%3F");
+    mapping.put('#', "%23");
+    mapping.put('[', "%5B");
+    mapping.put(']', "%5D");
+    mapping.put('@', "%40");
+    mapping.put('%', "%25");
+    if (toEncode == null) {
+      return null;
+    }
+    StringBuilder encoded = new StringBuilder(toEncode.length());
+    for (Character c : toEncode.toCharArray()) {
+      encoded.append(mapping.getOrDefault(c, c.toString()));
+    }
+    return encoded.toString();
+  }
+
+  public String getBaseUrl() {
+    return baseUrl;
+  }
+
+  public ImageRequest setBaseUrl(String baseUrl) {
+    this.baseUrl = baseUrl;
+    return this;
+  }
+
+  public String getRegion() {
+    return region;
+  }
+
+  public ImageRequest setRegion(String region) {
+    this.region = region;
+    return this;
+  }
+
+  public String getSize() {
+    return size;
+  }
+
+  public ImageRequest setSize(String size) {
+    this.size = size;
+    return this;
+  }
+
+  public String getRotation() {
+    return rotation;
+  }
+
+  public ImageRequest setRotation(String rotation) {
+    this.rotation = rotation;
+    return this;
+  }
+
+  public String getQuality() {
+    return quality;
+  }
+
+  public ImageRequest setQuality(String quality) {
+    this.quality = quality;
+    return this;
+  }
+
+  public String getExtension() {
+    return extension;
+  }
+
+  public ImageRequest setExtension(String extension) {
+    this.extension = extension;
+    return this;
+  }
+
+  /**
+   * Generates an IIIF Image Request URL of the ordering {scheme}://{server}{/prefix}/{identifier}/{region}/{size}/{rotation}/{quality}.{format} where each parameter is percent escaped
+   */
+  public String generateIIIFRequestUrl() {
+    String FORWARD_SLASH_DELIMITER = "/";
+    return baseUrl + FORWARD_SLASH_DELIMITER
+        + percentEncode(region)
+        + FORWARD_SLASH_DELIMITER
+        + percentEncode(size)
+        + FORWARD_SLASH_DELIMITER
+        + percentEncode(rotation)
+        + FORWARD_SLASH_DELIMITER
+        + percentEncode(quality)
+        + "."
+        + percentEncode(extension);
+  }
+
+  public void saveToFile(String filePath, String fileName) throws IOException {
+    URL url = new URL(this.generateIIIFRequestUrl());
+    BufferedImage img = ImageIO.read(url);
+    File file = new File(filePath + "/" + fileName + "." + this.getExtension());
+    ImageIO.write(img, this.getExtension(), file);
+    LOGGER.debug("Image downloaded and written to file successfully. Image url:\t" + url);
```

Comment:
In the logger, you can use '{}' as a placeholder for string interpolation, so the canonical way of doing this would be
```java
LOGGER.debug("Image downloaded and written to file successfully. Image url:\t{}", url);
```

Reply:
Done